### PR TITLE
 discovery-protocol handler

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: src/iden3comm/handlers/discovery-protocol.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+/**
+ * @beta
+ * createDiscoveryFeatureQueryMessage is a function to create didcomm protocol discovery-feature query message
+ * @param opts - discovery-feature query options
+ * @returns `DiscoverFeatureQueriesMessage`
+ */
+export function createDiscoveryFeatureQueryMessage(opts?: {
+  featureTypes?: (DiscoveryProtocolFeatureType | string)[];
+  from?: string;
```
Comment (Kolezhniuk) [inline src/iden3comm/handlers/discovery-protocol.ts:50]: should `from` and `to` be optional?

File: src/iden3comm/types/protocol/discovery-protocol.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/** @beta DiscoverFeatureQueriesMessage is struct the represents discover feature queries message */
+export type DiscoverFeatureQueriesMessage = BasicMessage & {
+  body: DiscoverFeatureQueriesBody;
+  type: typeof PROTOCOL_MESSAGE_TYPE.DISCOVERY_PROTOCOL_QUERIES_MESSAGE_TYPE;
+};
+
+/** @beta DiscoverFeatureQueriesBody is struct the represents discover feature queries body */
+export type DiscoverFeatureQueriesBody = {
+  queries: {
+    'feature-type': DiscoveryProtocolFeatureType | string;
```
Comment (Kolezhniuk) [inline src/iden3comm/types/protocol/discovery-protocol.ts:18]: Maybe better to declare  enum for that something like `DiscoverFeatureQueryType`? and use across all 'feature-type' occurences

Comment (Kolezhniuk) [reply src/iden3comm/types/protocol/discovery-protocol.ts:18]: Sould we allow to pass `DiscoveryProtocolFeatureType | string` arbitrary string here?

File: tests/handlers/payment.test.ts [hunk 3]
```diff
@@ -503,15 +503,15 @@ describe('payment-request handler', () => {
         tokenAddress: '0x71dcc8Dc5Eb138003d3571255458Bc5692a60eD4',
         recipient: '0xE9D7fCDf32dF4772A7EF7C24c76aB40E4A42274a',
         amount: '2',
-        expirationDate: '2024-12-04T18:43:31.082Z',
+        expirationDate: '2124-12-10T17:25:18.907Z',
```
Comment (Kolezhniuk) [inline tests/handlers/payment.test.ts:506]: looking in the btrigth future :)

**Review by Kolezhniuk — COMMENTED** [review]
package version bump?

File: src/iden3comm/handlers/discovery-protocol.ts (hunk 1 above)
Comment (volodymyr-basiuk) [reply src/iden3comm/handlers/discovery-protocol.ts:51]: I think yes, it's not mandatory for this message type

**Review by Kolezhniuk — APPROVED** [review]

**Review by vmidyllic — DISMISSED** [review]

File: src/iden3comm/handlers/discovery-protocol.ts [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    match?: string
+  ): DiscoverFeatureDisclosure[] {
+    if (!match || match === '*') {
+      return disclosures;
+    }
+    const regExp = this.wildcardToRegExp(match);
+    return disclosures.filter((disclosure) => regExp.test(disclosure.id));
+  }
+
+  private wildcardToRegExp(match: string): RegExp {
```
Comment (Kolezhniuk) [inline src/iden3comm/handlers/discovery-protocol.ts:270]: make const of regex, no need to recreate this every time on handle

**Review by Kolezhniuk — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**